src/test/java/org/opensearch/knn/index/FaissIT.java

```diff
@@ -1686,17 +1684,63 @@ public void testInvalidPQM_thenFail() {
         Map<String, Object> in = xContentBuilderToMap(xContentBuilder);
 
         createBasicKnnIndex(trainingIndexName, trainingFieldName, dimension);
-        ingestDataAndTrainModel(modelId, trainingIndexName, trainingFieldName, dimension, modelDescription, in, trainingDataCount);
-        assertTrainingFails(modelId, 360, 1000);
+        ResponseException re = expectThrows(
+            ResponseException.class,
+            () -> ingestDataAndTrainModel(modelId, trainingIndexName, trainingFieldName, dimension, modelDescription, in, trainingDataCount)
+        );
+        assertTrue(
+            re.getMessage().contains("Validation Failed: 1: parameter validation failed for MethodComponentContext parameter [encoder].;")
+        );
+    }
 
-        Response response = getModel(modelId, null);
+    @SneakyThrows
+    public void testIVF_InvalidPQM_thenFail() {
+        String trainingIndexName = "training-index";
+        String trainingFieldName = "training-field";
 
-        Map<String, Object> responseMap = createParser(
-            MediaTypeRegistry.getDefaultMediaType().xContent(),
-            EntityUtils.toString(response.getEntity())
-        ).map();
+        String modelId = "test-model";
+        String modelDescription = "test model";
 
-        assertEquals(KNNConstants.INVALID_CODE_COUNT_ERROR_MESSAGE, responseMap.get(MODEL_ERROR));
+        List<Integer> mValues = ImmutableList.of(16, 32, 64, 128);
+        int invalidPQM = 3;
+
+        // training data needs to be at least equal to the number of centroids for PQ
+        // which is 2^8 = 256.
+        int trainingDataCount = 256;
+
+        int dimension = testData.indexData.vectors[0].length;
+        SpaceType spaceType = SpaceType.L2;
+        int ivfNlist = 4;
+        int ivfNprobes = 4;
+        int pqCodeSize = 8;
+
+        XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()
```

Comment:
can you add json structure for below input? This will help to understand the structure. Thanks